### DISCUSSION TITLE
Accessing str member of Member

### DISCUSSION THREAD

**Discussion by hpux735** [diwic/dbus-rs#324] [Q&A] [answered]
I'm sorry if this is a very basic Rust question, but I'm trying to get to the contents of the `Member` of a DBus signal.  I've got it from the perspective of accessing the member property of a Message, and I can unwrap the optional, of course...  But getting the `str` from the Member struct is proving harder than I'd think...

I see the creation of the struct here: https://github.com/diwic/dbus-rs/blob/master/dbus/src/strings.rs#L194-L199

But, it looks like it only has a `.0` property that's private...  Is there an accessor I'm missing somewhere?  I couldn't find any examples in the docs for doing this.

**Comment by diwic** [comment, marked as answer]
As seen in [the documentation](https://docs.rs/dbus/0.9.2/dbus/strings/struct.Member.html), `Member` has a `Deref<Target=str>`, which means that you for the most part can use a `&Member` as if it were a `&str`. Example:

    let m = myMessage.member()?;
    let myStr: &str = &m; 
    
Or you can just use any `str` method on the member directly, like:

    let m = myMessage.member()?;
    let myStr = m.to_lowercase();

**Reply by hpux735** [reply]
Ok, thanks.
